app/models/ta.rb

```diff
@@ -15,13 +15,13 @@ class Ta < User
   has_many :grade_entry_student_tas
   has_many :grade_entry_students, through: :grade_entry_student_tas
 
-  def get_num_assigned(assignment)
-    assignment.ta_memberships.where(user_id: id).size
+  def get_num_assigned(assignment, ta_id = id)
+    assignment.ta_memberships.where(user_id: ta_id).size
   end
 
-  def get_num_marked(assignment)
+  def get_num_marked(assignment, ta_id = id)
```

Comment:
Make these methods only appear in the the `Assignment` model. Give it a signature `get_num_marked(assignment, ta_id = nil)` so that admins can call it without specifying an id and get everything, and tas can call it with their id to get the numbers specific to them.